rotors_gazebo_plugins/src/gazebo_ros_interface_plugin.cpp

```diff
@@ -0,0 +1,938 @@
+/*
+ * Copyright 2016 Geoffrey Hunter <gbmhunter@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <stdio.h>
+
+#include <boost/bind.hpp>
+#include <rotors_gazebo_plugins/gazebo_ros_interface_plugin.h>
+
+namespace gazebo {
+
+GazeboRosInterfacePlugin::GazeboRosInterfacePlugin()
+    : WorldPlugin(),
+      gz_node_handle_(0),
+      ros_node_handle_(0),
+      ros_actuators_msg_(new mav_msgs::Actuators) { }
+
+
+GazeboRosInterfacePlugin::~GazeboRosInterfacePlugin() {
+  event::Events::DisconnectWorldUpdateBegin(updateConnection_);
+}
+
+
+void GazeboRosInterfacePlugin::Load(physics::WorldPtr _world, sdf::ElementPtr _sdf) {
+
+  if(kPrintOnPluginLoad) {
+    gzdbg << __FUNCTION__ << "() called." << std::endl;
+  }
+
+  /// \brief    Store the pointer to the model.
+  world_ = _world;
+
+  namespace_.clear();
+
+  //==============================================//
+  //========== READ IN PARAMS FROM SDF ===========//
+  //==============================================//
+
+  if (_sdf->HasElement("robotNamespace"))
+    namespace_ = _sdf->GetElement("robotNamespace")->Get<std::string>();
+  else
+    gzerr << "Please specify a robotNamespace.\n";
+  gzdbg << "namespace_ = \"" << namespace_ << "\"." << std::endl;
+
+  /// @todo Fix this hack!!! Should not need this namespace set to firefly.
+  namespace_ = "firefly";
```

Comment:
I have fixed this. A namespace variable is no longer needed for the ROS interface plugin. Haven't checked it works with other MAVs yet.